### PR TITLE
Feat: add type selection in KV

### PR DESCRIPTION
**Pull request by weiy730**
Signed-off-by: weiy730 <weiy730@163.com>

### Description of your changes

Add type selection in KV, which can supprot to input and verificate different data types.

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [x] Run `yarn lint` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: src/extends/KV/index.tsx [hunk 1]
```diff
@@ -243,6 +301,13 @@ class KV extends Component<Props, State> {
                       />
                     </div>
                   </If>
+                  <If condition={item.isWrong == true}>
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:304]: You should use the form validator to check the value.

File: src/extends/KV/index.tsx [hunk 2]
```diff
@@ -216,13 +260,27 @@ class KV extends Component<Props, State> {
                   </If>
                 </Form.Item>
               </Col>
+              <Col span={5}>
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:263]: The span is too big. 10+5+10+1 > 24

File: src/extends/KV/index.tsx [hunk 3]
```diff
@@ -186,6 +229,7 @@ class KV extends Component<Props, State> {
     if (inputValue) {
       dataSource.push(inputValue);
     }
+    const options = [{ label: 'string', value: 'string' }, { label: 'number', value: 'number' }, { label: 'boolean', value: 'boolean' }, { label: 'object', value: 'object' }]
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:232]:
```suggestion
    const options = [{ label: 'String', value: 'string' }, { label: 'Number', value: 'number' }, { label: 'Boolean', value: 'boolean' }, { label: 'Object', value: 'object' }]
```

File: src/extends/KV/index.tsx [hunk 4]
```diff
@@ -216,13 +260,27 @@ class KV extends Component<Props, State> {
...
+              <Col span={5}>
+                <Form.Item>
+                  <Select
+                    showSearch={true}
+                    dataSource={options}
+                    {...init(`envType-${item.key}`)}
+                    label={'Type'}
+                    placeholder={i18n.t('Please select')}
+                    locale={locale().Select}
+                    onSearch={this.onSearch}
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:272]: Did not handle the value change?

File: src/extends/KV/index.tsx [hunk 5]
```diff
@@ -216,13 +260,27 @@ class KV extends Component<Props, State> {
...
+                  />
+                </Form.Item>
+              </Col>
               <Col span={10}>
                 <Form.Item>
-                  <If condition={item.valueType == 'number' || item.valueType == 'string'}>
+                  <If condition={item.valueType == 'number' || item.valueType == 'string' || item.valueType == 'object'}>
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:278]: You should provide the different forms for every type.

**Review by barnettZQG — CHANGES_REQUESTED** [review]

File: src/extends/KV/index.tsx (hunk 4 above)
Comment (weiy730) [reply src/extends/KV/index.tsx:272]: Like this ?

Comment (weiy730) [reply src/extends/KV/index.tsx:272]: If the type changes, the corresponding formItem will change too. Or this means something else ？

Comment (barnettZQG) [reply src/extends/KV/index.tsx:272]: It's great! Let's make the style better.

**Comment by weiy730** [comment]
I've fixed some of issues and committed the code

File: src/extends/KV/index.tsx [hunk 6]
```diff
@@ -186,84 +208,156 @@ class KV extends Component<Props, State> {
...
+                          placeholder={i18n.t('Please input a object')}
+                          {...init(`envValue-${item.key}`, {
+                            rules: [
+                              {
+                                validator: checkObjectFormat,
+                                message: 'Wrong format.',
+                              },
+                            ],
+                          })}
+                          value={item.value}
```
Comment (barnettZQG) [inline src/extends/KV/index.tsx:325]: We should format the value.

Comment (barnettZQG) [reply src/extends/KV/index.tsx:325]: OnChange:  string => object
Init Value: object => string

**Comment by barnettZQG** [comment]
This PR is not updated for a long time, I will close it.

**Pull request closed without merging**